examples/src/bin/ech-client.rs

```diff
@@ -205,9 +209,17 @@ struct Args {
 fn lookup_ech_configs(
     resolver: &Resolver,
     domain: &str,
+    port: u16,
 ) -> Option<pki_types::EchConfigListBytes<'static>> {
+    // For non-standard ports, lookup the ECHConfig using port-prefix naming
+    // See: https://datatracker.ietf.org/doc/html/rfc9460#section-9.1
+    let qname_to_lookup: String = match port {
+        443 => domain.to_string(),
+        other => format!("_{other}._https.{domain}").to_string(),
```

Comment:
A few small nits here:

1. We prefer to omit the `String` typing information if it can be inferred.
2. I think `to_string()` on the `format!()` result is superfluous - it's already a `String`.
3. (Minor) I would be tempted to keep using `port` as the binding name for the "other" arm. I think it will read a bit more clearly in the format string

```rust
    let qname_to_lookup = match port {
        443 => domain.to_string(),
        port => format!("_{port}._https.{domain}"),
    };
```

Reply:
Tiny nit: in cases like these where we're convertting from `&str` to `String` it can be helpful to use `to_owned()` instead of `to_string()`, which needs a little bit less work (because `ToString` depends on formatting machinery) and makes it clear that what's going on is more limited than conversion from anything to string.

Reply:
Thanks for the suggestions guys. It didn't even occur to me it would be "legal" to use the same name inside the arm, neat that it works.